pkg/kubelet/status/state/checkpoint.go

```diff
@@ -18,48 +18,62 @@ package state
 
 import (
 	"encoding/json"
-
 	"k8s.io/api/core/v1"
 	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager"
 	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/checksum"
 )
 
-var _ checkpointmanager.Checkpoint = &PodResourceAllocationCheckpoint{}
+var _ checkpointmanager.Checkpoint = &Checkpoint{}
```

Comment:
What's the purpose of this? Is it used anywhere?

Reply:
The same with `State` global variable in the beginning of the `checkpoint.go `

Reply:
To ensure struct `Checkpoint ` implement the interface `checkpointmanager.Checkpoint`.  Do you think it's unnecessary?

Reply:
It is unnecessary from my pov, but it's not related to this PR, so let's keep it as it is.